### DISCUSSION TITLE
how to set a datetime alarm to do something like repeated schedule?

### DISCUSSION THREAD

**Discussion by addy9908** [bonsai-rx/bonsai#1644] [Q&A]
Hi all,
I am new to Bonsai. I know how powerful it is, but I am a little lost, and need help:
I want to run something at exactly 5:00 pm everyday for 20 days. Once the set time meets
1. read from Arduono pin 2,3,4, then zip it with timestamp, and save as csv. This recording will stop 3hrs later
2. with a delay like 5s to wait for the signal read/csv save at step 1, sent out a 200ms trigger through another Arduino pin A0

then wait for another day when the set time meets, as something like a daily repeated event.

Thanks a lot.

BTW, It would be greatly appreciated if anyone can share me any website or courses or videos to learn the basic of Bonsai.

Best,
Zengyou

**Comment by bruno-f-cruz** [comment]
Re: title question, I would start with already existing examples and work from there. A quick search raised this one: https://github.com/orgs/bonsai-rx/discussions/1463

Regarding documentation, try the documentation website: [bonsai-rx.org/docs](https://bonsai-rx.org/docs/). Under the [Tutorial tab](https://bonsai-rx.org/docs/tutorials/acquisition.html) there are a lot of exercises you can run through.
Videos/documentation can also be found on the website: https://bonsai-rx.org/learn/ and on [youtube](https://www.youtube.com/watch?v=wPUCuSMCB-U)

**Reply by addy9908** [reply]
Hi Bruno,

Thank you so much for all the information. The first example is kind of similar to what I want, and I am sure I can learn a lot from there. I also appreciate all the learning sources!

Best,
Zengyou

**Reply by addy9908** [reply]
Hi Bruno,
Another question: where do you get the "Greater Than (TimeSpan)"? Do I need to install a specific package?
Best,
Zengyou

**Reply by bruno-f-cruz** [reply]
GreaterThan overload is created based on the type preceding it. If you have a DataTime object, adding a GreaterThan after it will create the operator you mentioned.

**Reply by addy9908** [reply]
Thank you so much Bruno. I am almost there to get it work now (have not try with Arduino yet).
Another question: I try to zip all 12 digital Inputs from Arduino, and save it with timestamp into one csv file. The problem is, if I zip more than 6 nodes, it will cause an error: "**Index was outside the bounds of the array**". I have already tried to zip every 6 first, then zip them together as 2 tuples, however the error is still there. Any suggestion how to zip more than 6 nodes?

Best,
Zengyou

**Comment by bruno-f-cruz** [comment]
As suggested in the pinned post. Please keep each discussions with its own
question and add the workflow you are working on. Fee free to open a new
discussion topic and someone will answer it there. Thanks!

**Reply by addy9908** [reply]
Thanks Bruno.
The zip question is a part of this project.  I tried the code again, and it may work now.
Here is my layout, and I will try to paste the code below:
![Arduino](https://github.com/bonsai-rx/bonsai/assets/11837676/217bc0ac-b7de-431a-82f3-1b0870e893ac)
<details>
  <summary>Workflow</summary>

> ```c#
> <?xml version="1.0" encoding="utf-8"?>
> <WorkflowBuilder Version="2.8.1"
>                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
>                  xmlns:ard="clr-namespace:Bonsai.Arduino;assembly=Bonsai.Arduino"
>                  xmlns:rx="clr-namespace:Bonsai.Reactive;assembly=Bonsai.Core"
>                  xmlns:io="clr-namespace:Bonsai.IO;assembly=Bonsai.System"
>                  xmlns:scr="clr-namespace:Bonsai.Scripting.Expressions;assembly=Bonsai.Scripting.Expressions"
>                  xmlns:viz="clr-namespace:Bonsai.Design.Visualizers;assembly=Bonsai.Design.Visualizers"
>                  xmlns:cv="clr-namespace:Bonsai.Vision;assembly=Bonsai.Vision"
>                  xmlns:ipy="clr-namespace:Bonsai.Scripting.IronPython;assembly=Bonsai.Scripting.IronPython"
>                  xmlns="https://bonsai-rx.org/2018/workflow">
>   <Workflow>
>     <Nodes>
>       <Expression xsi:type="Combinator">
>         <Combinator xsi:type="ard:CreateArduino">
>           <ard:Name>Arduino</ard:Name>
>           <ard:PortName>COM3</ard:PortName>
>           <ard:BaudRate>57600</ard:BaudRate>
>           <ard:SamplingInterval>10</ard:SamplingInterval>
>         </Combinator>
>       </Expression>
>       <Expression xsi:type="GroupWorkflow">
>         <Name>digitalInputs</Name>
>         <Workflow>
>           <Nodes>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED1" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>2</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED2" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>3</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED3" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>4</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED4" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>5</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED5" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>6</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED6" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>7</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="rx:Zip" />
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED7" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>8</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED8" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>9</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED9" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>10</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED10" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>11</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED12" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>13</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Pin" DisplayName="FED11" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalInput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>12</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="rx:Zip" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="rx:Zip" />
>             </Expression>
>             <Expression xsi:type="WorkflowOutput" />
>           </Nodes>
>           <Edges>
>             <Edge From="0" To="1" Label="Source1" />
>             <Edge From="1" To="12" Label="Source1" />
>             <Edge From="2" To="3" Label="Source1" />
>             <Edge From="3" To="12" Label="Source2" />
>             <Edge From="4" To="5" Label="Source1" />
>             <Edge From="5" To="12" Label="Source3" />
>             <Edge From="6" To="7" Label="Source1" />
>             <Edge From="7" To="12" Label="Source4" />
>             <Edge From="8" To="9" Label="Source1" />
>             <Edge From="9" To="12" Label="Source5" />
>             <Edge From="10" To="11" Label="Source1" />
>             <Edge From="11" To="12" Label="Source6" />
>             <Edge From="12" To="26" Label="Source1" />
>             <Edge From="13" To="14" Label="Source1" />
>             <Edge From="14" To="25" Label="Source1" />
>             <Edge From="15" To="16" Label="Source1" />
>             <Edge From="16" To="25" Label="Source2" />
>             <Edge From="17" To="18" Label="Source1" />
>             <Edge From="18" To="25" Label="Source3" />
>             <Edge From="19" To="20" Label="Source1" />
>             <Edge From="20" To="25" Label="Source4" />
>             <Edge From="21" To="22" Label="Source1" />
>             <Edge From="22" To="25" Label="Source5" />
>             <Edge From="23" To="24" Label="Source1" />
>             <Edge From="24" To="25" Label="Source6" />
>             <Edge From="25" To="26" Label="Source2" />
>             <Edge From="26" To="27" Label="Source1" />
>           </Edges>
>         </Workflow>
>       </Expression>
>       <Expression xsi:type="Combinator">
>         <Combinator xsi:type="rx:Timestamp" />
>       </Expression>
>       <Expression xsi:type="Disable">
>         <Builder xsi:type="io:CsvWriter">
>           <io:FileName>test\ZY_test_.csv</io:FileName>
>           <io:Append>false</io:Append>
>           <io:Overwrite>false</io:Overwrite>
>           <io:Suffix>Timestamp</io:Suffix>
>           <io:IncludeHeader>true</io:IncludeHeader>
>           <io:Selector>Timestamp,Value.Item1.Item1,Value.Item1.Item2,Value.Item1.Item3,Value.Item1.Item4,Value.Item1.Item5,Value.Item1.Item6,Value.Item2.Item1,Value.Item2.Item2,Value.Item2.Item3,Value.Item2.Item4,Value.Item2.Item5,Value.Item2.Item6</io:Selector>
>         </Builder>
>       </Expression>
>       <Expression xsi:type="rx:PublishSubject">
>         <Name>inputData</Name>
>       </Expression>
>       <Expression xsi:type="SubscribeSubject">
>         <Name>Timer</Name>
>       </Expression>
>       <Expression xsi:type="ExternalizedMapping">
>         <Property Name="Start" DisplayName="SessionStart" />
>       </Expression>
>       <Expression xsi:type="GroupWorkflow">
>         <Name>SessionStart</Name>
>         <Workflow>
>           <Nodes>
>             <Expression xsi:type="WorkflowInput">
>               <Name>Source1</Name>
>             </Expression>
>             <Expression xsi:type="MemberSelector">
>               <Selector>TimeOfDay</Selector>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Value" DisplayName="Start" />
>             </Expression>
>             <Expression xsi:type="GreaterThan">
>               <Operand xsi:type="TimeSpanProperty">
>                 <Value>PT12H55M</Value>
>               </Operand>
>             </Expression>
>             <Expression xsi:type="rx:Condition">
>               <Workflow>
>                 <Nodes>
>                   <Expression xsi:type="WorkflowInput">
>                     <Name>Source1</Name>
>                   </Expression>
>                   <Expression xsi:type="WorkflowOutput" />
>                 </Nodes>
>                 <Edges>
>                   <Edge From="0" To="1" Label="Source1" />
>                 </Edges>
>               </Workflow>
>             </Expression>
>             <Expression xsi:type="WorkflowOutput" />
>           </Nodes>
>           <Edges>
>             <Edge From="0" To="1" Label="Source1" />
>             <Edge From="1" To="3" Label="Source1" />
>             <Edge From="2" To="3" Label="Source2" />
>             <Edge From="3" To="4" Label="Source1" />
>             <Edge From="4" To="5" Label="Source1" />
>           </Edges>
>         </Workflow>
>       </Expression>
>       <Expression xsi:type="Combinator">
>         <Combinator xsi:type="rx:Take">
>           <rx:Count>1</rx:Count>
>         </Combinator>
>       </Expression>
>       <Expression xsi:type="ExternalizedMapping">
>         <Property Name="End" DisplayName="SessionEnd" />
>         <Property Name="FileName" DisplayName="csvFileName" />
>       </Expression>
>       <Expression xsi:type="rx:SelectMany">
>         <Name>SaveUntil</Name>
>         <Workflow>
>           <Nodes>
>             <Expression xsi:type="SubscribeSubject">
>               <Name>inputData</Name>
>             </Expression>
>             <Expression xsi:type="scr:ExpressionTransform">
>               <scr:Expression>new(
> Timestamp as TimeStamp,
> Timestamp.TimeOfDay.TotalMilliseconds as Millis,
> Value.Item1.Item1 as FED1,
> Value.Item1.Item2 as FED2,
> Value.Item1.Item3 as FED3,
> Value.Item1.Item4 as FED4,
> Value.Item1.Item5 as FED5,
> Value.Item1.Item6 as FED6,
> Value.Item2.Item1 as FED7,
> Value.Item2.Item2 as FED8,
> Value.Item2.Item3 as FED9,
> Value.Item2.Item4 as FED10,
> Value.Item2.Item5 as FED11,
> Value.Item2.Item6 as FED12
> )</scr:Expression>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="FileName" />
>             </Expression>
>             <Expression xsi:type="io:CsvWriter">
>               <io:FileName>test\ZY_FED_.csv</io:FileName>
>               <io:Append>false</io:Append>
>               <io:Overwrite>false</io:Overwrite>
>               <io:Suffix>Timestamp</io:Suffix>
>               <io:IncludeHeader>true</io:IncludeHeader>
>               <io:Selector />
>             </Expression>
>             <Expression xsi:type="viz:RollingGraphBuilder">
>               <viz:IndexSelector>TimeStamp</viz:IndexSelector>
>               <viz:ValueSelector>FED1,FED2,FED3,FED4,FED5,FED6,FED7,FED8,FED9,FED10,FED11,FED12</viz:ValueSelector>
>               <viz:SymbolType>None</viz:SymbolType>
>               <viz:LineWidth>1</viz:LineWidth>
>               <viz:Capacity>1000</viz:Capacity>
>               <viz:Min xsi:nil="true" />
>               <viz:Max xsi:nil="true" />
>             </Expression>
>             <Expression xsi:type="SubscribeSubject">
>               <Name>Timer</Name>
>             </Expression>
>             <Expression xsi:type="MemberSelector">
>               <Selector>TimeOfDay</Selector>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="Value" DisplayName="End" />
>             </Expression>
>             <Expression xsi:type="GreaterThan">
>               <Operand xsi:type="TimeSpanProperty">
>                 <Value>PT12H56M</Value>
>               </Operand>
>             </Expression>
>             <Expression xsi:type="rx:Condition">
>               <Workflow>
>                 <Nodes>
>                   <Expression xsi:type="WorkflowInput">
>                     <Name>Source1</Name>
>                   </Expression>
>                   <Expression xsi:type="WorkflowOutput" />
>                 </Nodes>
>                 <Edges>
>                   <Edge From="0" To="1" Label="Source1" />
>                 </Edges>
>               </Workflow>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="rx:Take">
>                 <rx:Count>1</rx:Count>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="WorkflowOutput" />
>           </Nodes>
>           <Edges>
>             <Edge From="0" To="1" Label="Source1" />
>             <Edge From="1" To="3" Label="Source1" />
>             <Edge From="2" To="3" Label="Source2" />
>             <Edge From="3" To="4" Label="Source1" />
>             <Edge From="5" To="6" Label="Source1" />
>             <Edge From="6" To="8" Label="Source1" />
>             <Edge From="7" To="8" Label="Source2" />
>             <Edge From="8" To="9" Label="Source1" />
>             <Edge From="9" To="10" Label="Source1" />
>             <Edge From="10" To="11" Label="Source1" />
>           </Edges>
>         </Workflow>
>       </Expression>
>       <Expression xsi:type="rx:SelectMany">
>         <Name>Wait for next day</Name>
>         <Workflow>
>           <Nodes>
>             <Expression xsi:type="SubscribeSubject">
>               <Name>inputData</Name>
>             </Expression>
>             <Expression xsi:type="MemberSelector">
>               <Selector>TimeStamp</Selector>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="rx:Skip">
>                 <rx:Count>1</rx:Count>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="rx:Zip" />
>             </Expression>
>             <Expression xsi:type="LessThan" />
>             <Expression xsi:type="rx:Condition">
>               <Workflow>
>                 <Nodes>
>                   <Expression xsi:type="WorkflowInput">
>                     <Name>Source1</Name>
>                   </Expression>
>                   <Expression xsi:type="WorkflowOutput" />
>                 </Nodes>
>                 <Edges>
>                   <Edge From="0" To="1" Label="Source1" />
>                 </Edges>
>               </Workflow>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="rx:Take">
>                 <rx:Count>1</rx:Count>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="WorkflowOutput" />
>           </Nodes>
>           <Edges>
>             <Edge From="0" To="1" Label="Source1" />
>             <Edge From="1" To="2" Label="Source1" />
>             <Edge From="1" To="3" Label="Source2" />
>             <Edge From="2" To="3" Label="Source1" />
>             <Edge From="3" To="4" Label="Source1" />
>             <Edge From="4" To="5" Label="Source1" />
>             <Edge From="5" To="6" Label="Source1" />
>             <Edge From="6" To="7" Label="Source1" />
>           </Edges>
>         </Workflow>
>       </Expression>
>       <Expression xsi:type="Combinator">
>         <Combinator xsi:type="rx:Repeat" />
>       </Expression>
>       <Expression xsi:type="ExternalizedMapping">
>         <Property Name="DueTime" DisplayName="TriggerDelay" />
>       </Expression>
>       <Expression xsi:type="Combinator">
>         <Combinator xsi:type="rx:Delay">
>           <rx:DueTime>PT0.1S</rx:DueTime>
>         </Combinator>
>       </Expression>
>       <Expression xsi:type="ExternalizedMapping">
>         <Property Name="TriggerDuration" />
>       </Expression>
>       <Expression xsi:type="GroupWorkflow">
>         <Name>A0TriggerOut</Name>
>         <Workflow>
>           <Nodes>
>             <Expression xsi:type="WorkflowInput">
>               <Name>Source1</Name>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="BooleanProperty">
>                 <Value>true</Value>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalOutput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>14</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="ExternalizedMapping">
>               <Property Name="DueTime" DisplayName="TriggerDuration" />
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="rx:Delay">
>                 <rx:DueTime>PT0.2S</rx:DueTime>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="BooleanProperty">
>                 <Value>false</Value>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="ard:DigitalOutput">
>                 <ard:PortName>Arduino</ard:PortName>
>                 <ard:Pin>14</ard:Pin>
>               </Combinator>
>             </Expression>
>             <Expression xsi:type="WorkflowOutput" />
>           </Nodes>
>           <Edges>
>             <Edge From="0" To="1" Label="Source1" />
>             <Edge From="1" To="2" Label="Source1" />
>             <Edge From="2" To="4" Label="Source1" />
>             <Edge From="3" To="4" Label="Source2" />
>             <Edge From="4" To="5" Label="Source1" />
>             <Edge From="5" To="6" Label="Source1" />
>             <Edge From="6" To="7" Label="Source1" />
>           </Edges>
>         </Workflow>
>       </Expression>
>       <Expression xsi:type="ExternalizedMapping">
>         <Property Name="FileName" DisplayName="AviName" />
>       </Expression>
>       <Expression xsi:type="Combinator">
>         <Combinator xsi:type="cv:CameraCapture">
>           <cv:Index>0</cv:Index>
>           <cv:CaptureProperties />
>         </Combinator>
>       </Expression>
>       <Expression xsi:type="Combinator">
>         <Combinator xsi:type="cv:VideoWriter">
>           <cv:FileName>test\ZY_FED_.avi</cv:FileName>
>           <cv:Suffix>Timestamp</cv:Suffix>
>           <cv:Buffered>true</cv:Buffered>
>           <cv:Overwrite>false</cv:Overwrite>
>           <cv:FourCC>FMP4</cv:FourCC>
>           <cv:FrameRate>30</cv:FrameRate>
>           <cv:FrameSize>
>             <cv:Width>0</cv:Width>
>             <cv:Height>0</cv:Height>
>           </cv:FrameSize>
>           <cv:ResizeInterpolation>NearestNeighbor</cv:ResizeInterpolation>
>         </Combinator>
>       </Expression>
>       <Expression xsi:type="ipy:PythonTransform">
>         <ipy:Name>frameCount</ipy:Name>
>         <ipy:Script>count = 0
> @returns(int)
> def process(value):
>   global count
>   count = count+1
>   return count</ipy:Script>
>       </Expression>
>       <Expression xsi:type="Combinator">
>         <Combinator xsi:type="rx:Timestamp" />
>       </Expression>
>       <Expression xsi:type="MemberSelector">
>         <Selector>Timestamp</Selector>
>       </Expression>
>       <Expression xsi:type="rx:PublishSubject">
>         <Name>Timer</Name>
>       </Expression>
>       <Expression xsi:type="GroupWorkflow">
>         <Name>FrameClock</Name>
>         <Workflow>
>           <Nodes>
>             <Expression xsi:type="WorkflowInput">
>               <Name>Source1</Name>
>             </Expression>
>             <Expression xsi:type="MemberSelector">
>               <Selector>Value</Selector>
>             </Expression>
>             <Expression xsi:type="MemberSelector">
>               <Selector>Timestamp.TimeOfDay.TotalMilliseconds</Selector>
>             </Expression>
>             <Expression xsi:type="Combinator">
>               <Combinator xsi:type="rx:Zip" />
>             </Expression>
>             <Expression xsi:type="scr:ExpressionTransform">
>               <scr:Expression>new(
> Item1 as FrameNumber,
> Item2 as Millis)</scr:Expression>
>             </Expression>
>             <Expression xsi:type="WorkflowOutput" />
>           </Nodes>
>           <Edges>
>             <Edge From="0" To="1" Label="Source1" />
>             <Edge From="0" To="2" Label="Source1" />
>             <Edge From="1" To="3" Label="Source1" />
>             <Edge From="2" To="3" Label="Source2" />
>             <Edge From="3" To="4" Label="Source1" />
>             <Edge From="4" To="5" Label="Source1" />
>           </Edges>
>         </Workflow>
>       </Expression>
>       <Expression xsi:type="io:CsvWriter">
>         <io:FileName>test\ZY_camera_.csv</io:FileName>
>         <io:Append>false</io:Append>
>         <io:Overwrite>false</io:Overwrite>
>         <io:Suffix>Timestamp</io:Suffix>
>         <io:IncludeHeader>true</io:IncludeHeader>
>       </Expression>
>     </Nodes>
>     <Edges>
>       <Edge From="1" To="2" Label="Source1" />
>       <Edge From="2" To="3" Label="Source1" />
>       <Edge From="3" To="4" Label="Source1" />
>       <Edge From="5" To="7" Label="Source1" />
>       <Edge From="6" To="7" Label="Source2" />
>       <Edge From="7" To="8" Label="Source1" />
>       <Edge From="8" To="10" Label="Source1" />
>       <Edge From="8" To="14" Label="Source1" />
>       <Edge From="9" To="10" Label="Source2" />
>       <Edge From="10" To="11" Label="Source1" />
>       <Edge From="11" To="12" Label="Source1" />
>       <Edge From="13" To="14" Label="Source2" />
>       <Edge From="14" To="16" Label="Source1" />
>       <Edge From="15" To="16" Label="Source2" />
>       <Edge From="17" To="19" Label="Source1" />
>       <Edge From="18" To="19" Label="Source2" />
>       <Edge From="18" To="20" Label="Source1" />
>       <Edge From="20" To="21" Label="Source1" />
>       <Edge From="21" To="22" Label="Source1" />
>       <Edge From="21" To="24" Label="Source1" />
>       <Edge From="22" To="23" Label="Source1" />
>       <Edge From="24" To="25" Label="Source1" />
>     </Edges>
>   </Workflow>
> </WorkflowBuilder>
> ``` 

</details>
